internal/db/model.go

```diff
@@ -75,6 +78,49 @@ func (d *Database) GetModel(ctx context.Context, model *dbmodel.Model) (err erro
 	return nil
 }
 
+// SetModelMigrationMode updates the model's migration mode to the provided value.
+// This function will return an error if the model's migration mode is already
+// set to anything besides MigrationModeNone. Use `UpdateModel` to complete the
+// migration and reset the migration mode to MigrationModeNone.
+func (d *Database) SetModelMigrationMode(ctx context.Context, uuid string, migrationMode dbmodel.MigrationMode) (m *dbmodel.Model, err error) {
```

Comment:
Does this need to return the model? I would've expected it to just set the migration mode and call it a day.